tools/automation/cli_linter/util.py

```diff
@@ -89,39 +87,34 @@ class LinterError(Exception):
     pass
 
 # return list of (cmd, params) tuples from example text. e.g. (az foo bar, [--opt, -n, -g, --help])
-def get_cmd_param_list_from_example(example_text):
-
-    def process_short_option(option): # handle options like -otable
-        if len(option) > 2 and option[0] == "-" and option[0] != option[1]:
-            return option[0:2]
-        else:
-            return option
-
-    CMD_PREFIX = "az "
-    commands = []  # some examples have multistep commands. This is a simple way to extract them
-    while(CMD_PREFIX in example_text):
-        # find next az command start and end
-        start = example_text.find(CMD_PREFIX)
-        end = example_text.find(CMD_PREFIX, start + 1)
-        end = end if end > -1 else len(example_text)
-        # extract command
-        cmd_text = example_text[start:end]
-        # update example text
-        example_text = example_text[end:]
-        # remove piping from cmd_text
-        end = cmd_text.find("|")
-        end = end if end > -1 else len(cmd_text)
-        cmd_text = cmd_text[:end]
-        # add to commands list
-        commands.append(cmd_text)
-
-
-    cmd_param_list = []
-    for command in commands:
-        idx = command.find(" -")  # Todo: positionals?
-        command_body = command[:idx].strip()
-        parameters = [maybe_opt for maybe_opt in command[idx:].split() if maybe_opt.startswith("-")]
-        parameters = list(map(process_short_option, parameters)) # process short options like -otable
-        cmd_param_list.append((command_body, parameters))
-
-    return cmd_param_list
+def extract_commands_from_example(example_text):
+
+    # fold commands spanning multiple lines into one line. Split commands that use pipes
+    example_text = example_text.replace("\\\n", " ")
```

Comment:
Will this not also combine separate commands into one?

Reply:
This Will not, it should only combine lines where the preceding line ends with a backslash and newline. To handle situations like [this](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py#L45-L56):

***
            az keyvault certificate create --vault-name vaultname -n cert1 \\
              -p "$(az keyvault certificate get-default-policy)"
            secrets=$(az keyvault secret list-versions --vault-name vaultname \\
              -n cert1 --query "[?attributes.enabled].id" -o tsv)
            vm_secrets=$(az vm secret format -s "$secrets")
            az vm create -g group-name -n vm-name --admin-username deploy  \\
              --image debian --secrets "$vm_secrets"
***